libqtile/widget/windowtabs.py

```diff
@@ -68,22 +62,17 @@ def _configure(self, qtile, bar):
     def update(self, *args):
         names = []
         for w in self.bar.screen.group.windows:
+            name = self.parse_text(w.name if w and w.name else " ")
             state = ""
             if w.maximized:
                 state = "[] "
             elif w.minimized:
                 state = "_ "
             elif w.floating:
                 state = "V "
-            task = "%s%s" % (state, w.name if w and w.name else " ")
-            task = pangocffi.markup_escape_text(task)
+            task = pangocffi.markup_escape_text(state + name)
             if w is self.bar.screen.group.current_window:
                 task = task.join(self.selected)
             names.append(task)
         self.text = self.separator.join(names)
-        if callable(self.parse_text):
-            try:
-                self.text = self.parse_text(self.text)
-            except:  # noqa: E722
-                logger.exception("parse_text function failed:")
```

Comment:
Please keep the `try... except...` here. `logger.exception` prints that line but then also prints the error traceback. Removing the logger line still logs the error but it's good to have that additional line first.

You'd want to move this to where you're now calling `parse_text`. You could also default to the window name if the function fails too.

Reply:
Got it! I've added the `try` block back and set a default value for the name in case of error.